simplematrixbotlib/match.py

```diff
@@ -111,26 +111,26 @@ def command(self, command=None):
             elif self.mention():
                 # the order is important here!
                 # note: we assume that body and formatted_body, if present, match as a workaraound of cleaning html
-                if self.event.body.startswith(self._disambiguated_name):
-                    self._body_without_prefix = self.event.body[len(self._disambiguated_name):]
-                elif self.event.body.startswith(self._display_name):
-                    self._body_without_prefix = self.event.body[len(self._display_name):]
-                elif self.event.body.startswith(self.room.own_user_id):
-                    self._body_without_prefix = self.event.body[len(self.room.own_user_id):]
-                elif self.event.formatted_body.startswith(self._pill):
-                    name = self.event.formatted_body[len(self._pill):self.event.formatted_body.index('</a>')]
+                id_matched = False
+                for id in (self._disambiguated_name, self._display_name, self.room.own_user_id):
```

Comment:
assuming `_disambiguated_name` doesn't match, do we want to allow `_display_name` to match?
also, I think we need to look closer at delimiting space/colon.
currently, if there is a bot "Krazy" and a user "KrazyKirby99999", then a message starting in the latter is matched by the former.

Reply:
I think that we should restrict the criteria for mentioning. How about reducing it to the matrix.to link and the user_id?
(...matrix.to...) and (@user:home.server)

Reply:
I added these 4 variants to support all common cases:
- pill is inserted when clicking or tab-completing in element web
- MXID seems like it is the real thing
- I don't know where it comes from, but I think some clients like maybe mobile insert displayname instead of pill for unknown reason. or maybe fluffychat, nheko, ?
- if done properly the former would be disambiguated when there are clashes